presto-clickhouse/src/main/java/com/facebook/presto/plugin/clickhouse/DateTimeUtil.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.clickhouse;
+
+import java.sql.Date;
+import java.sql.Time;
+import java.time.DateTimeException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoField;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public final class DateTimeUtil
+{
+    private DateTimeUtil()
+    {
+    }
+
+    public static Date convertZonedDaysToDate(long zonedDays)
+    {
+        long epochMillis;
+
+        try {
+            ZonedDateTime zonedDateTime = ZonedDateTime.of(LocalDate.ofEpochDay(zonedDays), LocalTime.MIN, ZoneId.systemDefault());
+            epochMillis = SECONDS.toMillis(zonedDateTime.toEpochSecond());
+        }
+        catch (DateTimeException ex) {
+            epochMillis = zonedDays < 0 ? Long.MIN_VALUE : Long.MAX_VALUE;
+        }
+
+        return new Date(epochMillis);
+    }
+
+    public static long convertDateToZonedDays(Date date)
+    {
+        try {
+            ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(date.getTime()), ZoneId.systemDefault());
```

Comment:
`getMillisKeepLocal` method in Joda is a bit confusing with its naming and its javadoc comment... What it does is converting milliseconds in given zone as first parameter of method, to milliseconds at instance's time zone.

If we run test below in a time zone with negative offset (pacific time, eastern time, etc), it passes (local time zone is important for the test because we make smaller/greater comparison between milliseconds):  
```
long oneUTCDayMillis = 24 * 60 * 60 * 1000;
DateTimeZone localZone = ISOChronology.getInstance().getZone();
System.out.println("Local zone: " + localZone);

long localMillis = localZone.getMillisKeepLocal(UTC, oneUTCDayMillis);
assertTrue(oneUTCDayMillis > localMillis);
System.out.println("UTC millis: " + oneUTCDayMillis + " > local millis: " + localMillis);

DateTimeZone utcZone = ISOChronology.getInstanceUTC().getZone();
System.out.println("UTC zone: " + utcZone);

long utcMillis = utcZone.getMillisKeepLocal(localZone, localMillis);
assertEquals(oneUTCDayMillis, utcMillis);
System.out.println("UTC millis: " + oneUTCDayMillis + " == new UTC millis: " + utcMillis);
```

The output for me is:  
```
Local zone: America/Bahia_Banderas
UTC millis: 86400000 > local millis: 61200000
UTC zone: UTC
UTC millis: 86400000 == new UTC millis: 86400000
```

I believe there are issues with naming in the previous code:  
```
long localMillis = resultSet.getDate(columnIndex).getTime();
long utcMillis = ISOChronology.getInstance().getZone().getMillisKeepLocal(UTC, localMillis);
return MILLISECONDS.toDays(utcMillis);
```

What happens here actually converting UTC millis to local millis, then converting local millis to days.